src/main/java/com/redhat/devtools/intellij/tektoncd/listener/SaveInEditorListener.java

```diff
@@ -0,0 +1,138 @@
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.dsl.base.CustomResourceDefinitionContext;
+import org.apache.commons.lang.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+import static com.redhat.devtools.intellij.tektoncd.Constants.KIND_PLURAL;
+import static com.redhat.devtools.intellij.tektoncd.Constants.NOTIFICATION_ID;
+
+public class SaveInEditorListener extends FileDocumentSynchronizationVetoer {
+    Logger logger = LoggerFactory.getLogger(SaveInEditorListener.class);
+
+    @Override
+    public boolean maySaveDocument(@NotNull Document document, boolean isSaveExplicit) {
+        VirtualFile vf = FileDocumentManager.getInstance().getFile(document);
+
+        // if file is not related to tekton we can skip it
+        if (vf == null || vf.getUserData(KIND_PLURAL).isEmpty()) {
+            return true;
+        }
+
+        Notification notification;
+        String namespace, name, apiVersion;
+        JsonNode spec;
+        CustomResourceDefinitionContext crdContext;
+        try {
+            namespace = YAMLHelper.getStringValueFromYAML(document.getText(), new String[] {"metadata", "namespace"});
+            name = YAMLHelper.getStringValueFromYAML(document.getText(), new String[] {"metadata", "name"});
+            if (Strings.isNullOrEmpty(namespace) || Strings.isNullOrEmpty(name)) {
+                throw new IOException("Tekton file has not a valid format. Namespace and/or name fields are invalid.");
+            }
+            apiVersion = YAMLHelper.getStringValueFromYAML(document.getText(), new String[] {"apiVersion"});
+            if (Strings.isNullOrEmpty(apiVersion)) {
+                throw new IOException("Tekton file has not a valid format. ApiVersion field is not found.");
+            }
+            crdContext = CRDHelper.getCRDContext(apiVersion, vf.getUserData(KIND_PLURAL));
+            if (crdContext == null) {
+                throw new IOException("Tekton file has not a valid format. ApiVersion field contains an invalid value.");
+            }
+            spec = YAMLHelper.getValueFromYAML(document.getText(), new String[] {"spec"});
+            if (spec == null) {
+                throw new IOException("Tekton file has not a valid format. Spec field is not found.");
+            }
+        } catch (IOException e) {
+            notification = new Notification(NOTIFICATION_ID, "Error", "An error occurred while saving \n" + e.getLocalizedMessage(), NotificationType.ERROR);
+            Notifications.Bus.notify(notification);
+            logger.error("Error: " + e.getLocalizedMessage());
+            return false;
+        }
+
+        int resultDialog = UIHelper.executeInUI(() ->
+                Messages.showYesNoDialog(
+                        "The file has been saved. Do you want to push the changes to the cluster?",
```

Comment:
Shouldn't we change this label as the file may not be saved depending on the user response ?

Reply:
Yes. What do you suggest?

Reply:
Having "Do you want to push the changes to the cluster?" is enough?

Reply:
Changed it so i can merge and rebase other PRs. If you have a better idea we can change it in the other PR related to save action